_posts/2023-07-05-niklas-fuzzing.md

```diff
@@ -0,0 +1,85 @@
+---
+title: "Niklas Gögge on Fuzz Testing"
+permalink: /blog/2023/07/05/fuzzing/
+layout: post
+author: brink
+name: "brink"
+alt: Niklas Gögge on Fuzz Testing
+category: "News"
+description: Niklas Gögge on Fuzz Testing.
+---
+
+In a security-focused project like Bitcoin Core, testing is critical. [Bitcoin
+Core unit tests][] are written in C++ using the Boost libraries to quickly test
+components of the codebase in isolation (see the [bech32 unit tests][] for
+example). [Bitcoin Core functional tests][] are written in python to test
+features of the software, involving multiple components working together (see
+the [P2P ping functional tests][]).
+
+A more cutting-edge type of testing is fuzz testing where software repeatedly calls
+into portions of the Bitcoin Core codebase with random, potentially
+malformed or semi-valid inputs in order to surface potential negative consequences
+such as a crash, memory leak, or other failure scenarios. When done
+continuously, these auto-generated mutated inputs from the fuzzing engine can be
+used to expand the test coverage and uncover bugs.
+
```

Comment:
I think we can deduplicate a bit and combine this with the subsequent paragraph.
```suggestion
into portions of the codebase using inputs generated automatically by a fuzzing engine
instead of manually by the test writer. When done
continuously, fuzzing can be
used to expand test coverage and uncover bugs.
```